src/build/env.py

```diff
@@ -169,20 +176,28 @@ def _create_isolated_env_venv(path):  # type: (str) -> Tuple[str, str]
     import venv
 
     venv.EnvBuilder(with_pip=True).create(path)
-    executable, script_dir = _find_executable_and_scripts(path)
-    # avoid the setuptools from ensurepip to break the isolation
-    if sys.version_info < (3, 6, 6):  # python 3.5 up to 3.6.5 come with pip 9 that's too old, for new standards
+    executable, script_dir, purelib = _find_executable_and_scripts(path)
+
+    # Get the version of pip in the environment
+    pip_distribution = next(iter(metadata.distributions(name='pip', path=[purelib])))
```

Comment:
Isn't the return value of `metadata.distributions` an iterator or why are we casting it to `iter`?

Reply:
Because it's an `Iteratable`, not an `Iterator`. You have to call `iter()` to get an `Iterator` that you call next with. Yes, it happens to work at runtime to call next on it, because the "Iterator" it returns happens to be itself, but it's technically wrong, and MyPy knows this.

In "real life", you never call `next()` without calling `iter()` first.

Reply:
Chain results are iterators.  If mypy thinks it's not an iterator, `distributions` is underspecified in the typeshed.

Reply:
It looks like these are correct here: https://github.com/python/typeshed/blob/master/stdlib/itertools.pyi

Reply:
However, this is returning an Iterable, for example, not an Iterator: https://github.com/python/typeshed/blob/27f38b8c23da0a2d0ff9ce486f84664c8b9a2d58/stdlib/importlib/metadata.pyi#L41

Reply:
Indeed, but it might be intentional, so that people don't rely on it being an iterator going forward.

Reply:
Let's keep it the way it is :)

Reply:
Where is `importlib<underscore>metadata` typed? Is that same file reused somehow?

Reply:
Okay. Up to you if you think it's worth asking about.